builder/ebs/step_enable_deprecation.go

```diff
@@ -22,7 +22,12 @@ func (s *stepEnableDeprecation) Run(ctx context.Context, state multistep.StateBa
 	}
 
 	ec2conn := state.Get("ec2").(*ec2.EC2)
-	amis := state.Get("amis").(map[string]string)
+	amis, ok := state.Get("amis").(map[string]string)
+	if !ok {
+		ui.Error("no AMIs to depreciate")
+		ui.Error("make sure you did not include the `skip_create_ami' directive in the build")
+		return multistep.ActionHalt
```

Comment:
A hard failure if no amis are in state sounds right. I think we can error on the type conversion error, and provide a check higher up specifically for the AMISkipCreateImage. 

```
// See builder/common/step_ami_region_copy.go for an example on how this information is being passed in. 
 if s.AMISkipCreateImage {
 69     ui.Say("Skipping Enable AMI deprecation because skip_create_ami is true")
 70     return multistep.ActionContinue
 71   }
```


```suggestion
		err := fmt.Errorf("no AMIs found in state to depreciate")
		state.Put("error", err)
		ui.Error(err.Error()
		return multistep.ActionHalt
```
By putting error into the state Packer will bubble the error upstream and include it as a final error for the failed build. Which will ensure it gets displayed as the cause for the failed build.

Reply:
I agree with the check higher up, so the rest of the build can continue if the command is invoked with skip_ami for testing purposes.
I've implemented your suggestion, I'll get to the check in a separate commit